File: frontend/src/types/project.ts
export interface Project {
  id: string;
  user_id?: string;
  name: string;
  description?: string;
  spec?: string;
  workflow_id?: string;
  config_overrides?: Record<string, Record<string, unknown>>;
  status: 'active' | 'archived' | 'deleted';
  total_tasks?: number;
  completed_tasks?: number;
  total_tokens?: number;
  total_cost?: number;
  created_by?: string;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/project.py
"""Project-related Pydantic schemas."""
from typing import Optional

from pydantic import BaseModel


class ProjectCreate(BaseModel):
    name: str
    description: Optional[str] = None
    spec: Optional[str] = None
    workflow_id: Optional[str] = None
    config_overrides: Optional[dict] = None


class ProjectUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    spec: Optional[str] = None
    status: Optional[str] = None
    config_overrides: Optional[dict] = None


class ProjectOut(BaseModel):
    id: str
    user_id: str
    name: str
    description: Optional[str] = None
    spec: Optional[str] = None
    workflow_id: Optional[str] = None
    config_overrides: Optional[dict] = None
    status: str = "active"
    total_tasks: int = 0
    completed_tasks: int = 0
    total_tokens: int = 0
    total_cost: float = 0.0
    created_at: str = ""
    updated_at: str = ""

    model_config = {"from_attributes": True}
